management-core/src/main/java/com/ns/task/service/ProductCoreServiceImpl.java

```diff
@@ -38,15 +47,23 @@ public ProductCoreServiceImpl(ModelMapper mapper, ProductRepository repository)
                 .collect(Collectors.toList());
     }
 
+    @RabbitListener(queues = "q.management.insert")
+    public ProductDTO receiverRPC(ProductDTO product) {
+        logger.info("Received message from RabbitMQ: {}", product.toString());
+        return insertProduct(product);
+    }
+
     @Override
     public ProductDTO insertProduct(ProductDTO product) {
         ProductEntity productEntity = convertToEntity(product);
         try {
             logger.debug("Sending data to DB {}", productEntity);
             productEntity = repository.saveProduct(productEntity);
+            product.setMessage("Name and model of the service exists already");
             logger.debug("Getting persisted data from insert to DB {}", productEntity);
         } catch (DataIntegrityViolationException exception) {
-            throw new ProductManagementException("Name and model of the product exists already", exception);
+            product.setMessage("Name and model already exists");
+            return product;
         }
         return convertToDto(productEntity);
```

Comment:
At the beginning of the method you convert a DTO to an Entity, and at the end, you convert an Entity to a DTO, it's kind of weird and I think it's unnecessary these mappings.